reference-adapters/ifms-adapter/src/main/java/org/egov/utils/MdmsUtils.java

```diff
@@ -138,4 +139,21 @@ public MdmsCriteriaReq prepareMdMsRequestWithActiveFilter(RequestInfo requestInf
 				.build();
 	}
 
+	public Boolean validateFromAndToDates (JsonNode jsonNode, String detailsArrayName) {
+		JsonNode detailsArray = jsonNode.get(detailsArrayName);
+		if (detailsArray != null && detailsArray.isArray()) {
+			for (JsonNode details : detailsArray) {
+				long effectiveFrom = details.get("effectiveFrom").asLong();
+				Long effectiveTo = details.has("effectiveTo") ?
+						details.get("effectiveTo").asLong() : null;
+				long currentTime = System.currentTimeMillis();
+
+				if (effectiveFrom > currentTime || (effectiveTo != null && effectiveTo < currentTime)) {
```

Comment:
@shubhang-eGov shouldn't this be an "&" instead of OR? If effectiveTo is present, then we need to validate both effectiveFrom and to. If effectiveTo is not there, we can assume that it is effective as of today. Can this block be simplified?